runtime/vm/jvminit.c

```diff
@@ -1823,19 +1824,41 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 				IDATA argIndexCompactOnIdleDisable = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXIDLETUNINGCOMPACTONIDLEDISABLE, NULL);
 				IDATA argIndexIgnoreUnrecognizedOptionsEnable = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXIDLETUNINGIGNOREUNRECOGNIZEDOPTIONSENABLE, NULL);
 				IDATA argIndexIgnoreUnrecognizedOptionsDisable = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXIDLETUNINGIGNOREUNRECOGNIZEDOPTIONSDISABLE, NULL);
+				BOOLEAN enableGcOnIdle = FALSE;
+				BOOLEAN inContainer = FALSE;
+				int32_t errorCode = 0;
+				
+				inContainer = omrsysinfo_is_running_in_container(&errorCode);
+				inContainer = ((0 == errorCode) && (TRUE == inContainer));
```

Comment:
It is not clear from the API what is the return value when error happens. During error condition, it could return TRUE or FALSE. That is why the explicit check for error code.

Reply:
I've checked all the implementations and opened a PR at OMR to clarify the functions contract: https://github.com/eclipse/omr/pull/3380

The function is most useful it returns a consistent answer (FALSE) on an error and that's what it does.

I think we can drop checking the error code here.

Reply:
Ok, thanks for clarifying the contract. I have updated the change to remove the error check.